registry/registry_ui.go

```diff
@@ -512,7 +539,73 @@ func createUpdateNamespace(ctx *gin.Context, isUpdate bool) {
 					Msg:    "You don't have permission to modify an approved registration. Please contact your federation administrator"})
 				return
 			}
+
+			// If non-admin user accesses a namespace with user_id != user but with access_token
+			if !isAdmin && !belongsTo && accessToken != "" {
+				jwks, err := jwk.Parse([]byte(ns.Pubkey))
+				if err != nil {
+					log.Errorf("Error parsing the public key of the namespace %s with ID %d: %v", ns.Prefix, ns.ID, err)
+					ctx.JSON(http.StatusInternalServerError, server_structs.SimpleApiResp{
+						Status: server_structs.RespFailed,
+						Msg:    fmt.Sprintf("Error parsing the public key of the namespace %s with ID %d: %v", ns.Prefix, ns.ID, err),
+					})
+					return
+				}
+
+				accessJWT, err := jwt.Parse(
+					[]byte(accessToken),
+					jwt.WithKeySet(jwks),
+					jwt.WithVerify(true),
+				)
+				if err != nil {
+					ctx.JSON(http.StatusForbidden,
+						server_structs.SimpleApiResp{
+							Status: server_structs.RespFailed,
+							Msg:    fmt.Sprint("Invalid access token: ", err),
+						})
+					return
+				}
+
+				scopeValidator := token_scopes.CreateScopeValidator([]token_scopes.TokenScope{token_scopes.Registry_EditRegistration}, false)
+				if err = jwt.Validate(accessJWT, jwt.WithValidator(scopeValidator)); err != nil {
+					log.Errorf("Failed to verify the scope of the token: %v", err)
```

Comment:
Should these logged errors also enforce a lowercase? Or do we only care about then when passing errors back up the error stack?

Reply:
The logged error are not chained together (unless the real Go error), so I don't think we should put them in lower case if they are meant to be a complete sentence.